### PR TITLE
Handle or proxy all

### PR DESCRIPTION
**Pull request by markyen**
Takes a request with multiple keys and fans it out to all the destinations responsible

@jwolski @Raynos

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "view-cover": "opn coverage/index.html"
   },
   "dependencies": {
-    "body": "^4.6.0",
+    "body": "Raynos/body#5308bb4",
```
Comment (Raynos) [inline package.json]: we cannot ship github links into prod. We'll need to wait for a new version of body.

**Comment by Raynos** [comment]
lgtm. Let's wait for body though.

File: index.js [hunk 2]
```diff
@@ -841,6 +842,37 @@ RingPop.prototype.handleOrProxy =
...
+                process.nextTick(self.proxyReq.bind(self, _.defaults({
+                    dest: dest,
+                    res: res
+                }, opts)));
+            }
+            reses[dest] = res;
+        });
+        return reses;
+
+        function mapUniq(list, iteratee) {
```
Comment (jwolski) [inline index.js]: if I were to channel my inner Matt Ranney, can you extract this method to ringpop's prototype to reduce GC?

File: index.js [hunk 3]
```diff
@@ -841,6 +842,37 @@ RingPop.prototype.handleOrProxy =
...
+        var self = this;
+        var keys = opts.keys;
+        var req = opts.req;
+        var localHandler = opts.localHandler;
+
+        var reses = {};
+        var dests = mapUniq(keys, this.lookup.bind(this));
+        dests.forEach(function(dest) {
+            var res = hammock.Response();
+            if (self.whoami() === dest) {
```
Comment (jwolski) [inline index.js]: extract this from the loop and call whoami() only once

File: index.js [hunk 4]
```diff
@@ -841,6 +842,37 @@ RingPop.prototype.handleOrProxy =
...
+                process.nextTick(localHandler.bind(null, req, res));
+            } else {
+                process.nextTick(self.proxyReq.bind(self, _.defaults({
+                    dest: dest,
+                    res: res
+                }, opts)));
```
Comment (jwolski) [inline index.js]: is there a reason not to be explicit with the opts that we're sending to proxyReq and just providing all of `opts`? This'll pass over `localHandler` which is obviously unnecessary.

**Comment by jwolski** [comment]
seems legit. minor comments left

File: index.js [hunk 5]
```diff
@@ -841,6 +842,37 @@ RingPop.prototype.handleOrProxy =
...
+RingPop.prototype.handleOrProxyAll =
+    function handleOrProxyAll(opts) {
+        var self = this;
+        var keys = opts.keys;
+        var req = opts.req;
+        var localHandler = opts.localHandler;
+
+        var reses = {};
+        var dests = mapUniq(keys, this.lookup.bind(this));
+        dests.forEach(function(dest) {
```
Comment (jwolski) [inline index.js]: looks like a reasonable use case for:

```
var reses = dests.reduce(...);
```

ill leave it up to you.

File: index.js (hunk 2 above)
Comment (Raynos) [inline index.js]: `mapUniq` doesn't need `self` or any closure variables. Just put it in the file as a top level function declaration, no need to be on the prototype.

Comment (jwolski) [inline index.js]: that's true too

File: index.js (hunk 5 above)
Comment (markyen) [inline index.js]: I like that better too. Fixed

File: index.js (hunk 3 above)
Comment (markyen) [inline index.js]: Fixed

File: index.js (hunk 4 above)
Comment (markyen) [inline index.js]: Fixed

File: index.js (hunk 2 above)
Comment (markyen) [inline index.js]: Fixed

**Comment by jwolski** [comment]
looks great. feel free to self-merge and then plz minor version bump and push tag. thx

**Pull request merged**